src/components/AddressHistory.js

```diff
@@ -0,0 +1,352 @@
+   */
+  updateListWs = (tx) => {
+    // We only add new tx/blocks if it's the first page
+    if (!this.state.hasBefore) {
+      if (this.props.shouldUpdate(tx, true)) {
+        let transactions = this.state.transactions;
+        let hasAfter = (this.state.hasAfter || (transactions.length === TX_COUNT && !this.state.hasAfter))
+        transactions = helpers.updateListWs(transactions, tx, TX_COUNT);
+
+        let firstHash = transactions[0].tx_id;
+        let lastHash = transactions[transactions.length-1].tx_id;
+
+        const newNumberOfTransactions = this.state.numberOfTransactions + 1;
+
+        // Finally we update the state again
+        this.setState({ transactions, hasAfter, firstHash, lastHash, numberOfTransactions: newNumberOfTransactions });
+        if (this.props.addressSummaryRef.current) {
+          this.props.addressSummaryRef.current.updateNumberOfTransactions(newNumberOfTransactions);
+        }
+      }
+    }
+  }
+
+  /**
+   * Update component state when new list data arrives
+   *
+   * @param {Object} data Response data from the server
+   * @param {Object} queryParams URL parameters
+   */
+  handleDataFetched = (data, queryParams) => {
+    // Handle differently if is the first GET response we receive
+    // page indicates if was clicked 'previous' or 'next'
+    // Set first and last hash of the transactions
+    let firstHash = null;
+    let lastHash = null;
+    if (data.transactions.length) {
+      firstHash = data.transactions[0].tx_id;
+      lastHash = data.transactions[data.transactions.length-1].tx_id;
+    }
+
+    let hasAfter;
+    let hasBefore;
+    if (queryParams.page === 'previous') {
+      hasAfter = true;
+      hasBefore = data.has_more;
+      if (!hasBefore) {
+        // Went back to most recent page: clear URL params
+        this.props.pagination.clearOptionalQueryParams();
+      }
+    } else if (queryParams.page === 'next') {
+      hasBefore = true;
+      hasAfter = data.has_more;
+    } else {
+      // First load without parameters
+      hasBefore = false;
+      hasAfter = data.has_more;
+    }
+
+    this.setState({
+      transactions: data.transactions,
+      loading: false,
+      firstHash,
+      lastHash,
+      hasAfter,
+      hasBefore,
+      queryParams,
+      numberOfTransactions: data.total,
+    });
+
+    if (this.props.addressSummaryRef.current) {
+      this.props.addressSummaryRef.current.updateNumberOfTransactions(data.total);
+    }
+  }
+
+  /**
+   * Update transactions data state after requesting data from the server
+   *
+   * @param {Object} queryParams URL parameters
+   */
+  getData = (queryParams) => {
+    hathorLib.walletApi.getSearchAddress(this.props.address, TX_COUNT, queryParams.hash, queryParams.page, queryParams.token, (response) => {
+      if (response.success) {
+        this.handleDataFetched(response, queryParams);
+      }
+    });
+  }
+
+  /**
+   * We get the final balance for the search address and selected token
```

Comment:
Took me a while to understand this docstring. It's important to say it's _one tx's_ balance, not the address total balance. When I read "the final balance for the search address" I imagined the total balance.

Reply:
Makes sense. Updated the docstring